public/app/core/time_series2.ts

```diff
@@ -8,8 +9,12 @@ function matchSeriesOverride(aliasOrRegex, seriesAlias) {
   }
 
   if (aliasOrRegex[0] === '/') {
-    const regex = kbn.stringToJsRegex(aliasOrRegex);
-    return seriesAlias.match(regex) != null;
+    try {
+      const regex = kbn.stringToJsRegex(aliasOrRegex);
+      return seriesAlias.match(regex) != null;
+    } catch (e) {
+      return appEvents.emit('alert-error', ['Invalid aliasOrRegex value.', e.message]);
```

Comment:
Is there a way to thow an error here instead, don't want this low level model to issue an app level UI event

Reply:
Yep. Probably let the error bubble up to the (graph) panel level and handle it there. 

Thanks for the feedback. I'll have a go again tomorrow morning.

Reply:
@torkelo I have reworked the solution and squash them as 1 commit.

the lower level module `time_series2.ts` no longer catches the exception. Instead the error from the util lib is allowed to bubble up to the graph panel level where it is handled there. 

Please review and let me know if the change is alright.

Reply:
time_series 2 is also a low level model component that should not send messages

Reply:
ok thnks for the feedback will try again

Reply:
@torkelo Sorry for the late of response, was working on another PR and busy with work.

I have rework on the PR based on your recommendation. The `try-catch` block is now on the controller level. Hopefully it is ok now. Let me know if there is still issue with it. Happy to improve it.